apollo-federation/src/sources/connect/fetch_dependency_graph/mod.rs

```diff
@@ -757,6 +986,451 @@ mod tests {
+            let source_entering_edge = *source_entry_edges.last().unwrap();
+            let response_name = name!("_merge_nested_path_test");
+
+            let (unmatched, selection) = JSONSelection::parse(
+                ".foo.bar {
+                    qux: .qaax
+                    qax: .qaax {
+                      baz
+                    }
+                }",
+            )
+            .unwrap();
+            assert!(unmatched.is_empty());
+
+            let mut node = source::fetch_dependency_graph::Node::Connect(
+                connect::fetch_dependency_graph::Node {
+                    merge_at: merge_at.clone(),
+                    source_entering_edge,
+                    field_response_name: response_name.clone(),
+                    field_arguments: arguments.clone(),
+                    selection: Some(selection),
+                },
+            );
+
+            fetch_graph
+                .add_path(
+                    query_graph,
+                    source::fetch_dependency_graph::Path::Connect(
+                        connect::fetch_dependency_graph::Path {
+                            merge_at,
+                            source_entering_edge,
+                            source_id,
+                            field: Some(connect::fetch_dependency_graph::PathField {
+                                response_name,
+                                arguments,
+                                selections:
+                                    connect::fetch_dependency_graph::PathSelections::Selections {
+                                        head_property_path: vec![
+                                            Key::Field("foo".to_string()),
+                                            Key::Field("bar".to_string()),
+                                        ],
+                                        named_selections: vec![
+                                            // - Node's selection: a b { x z { three } } c: last
+                                            (name!("qax"), vec![Key::Field("qaax".to_string())]),
+                                            (
+                                                name!("baaz"),
+                                                vec![
+                                                    Key::Field("baz".to_string()),
+                                                    Key::Field("buzz".to_string()),
+                                                ],
+                                            ),
+                                        ],
+                                        tail_selection: Some((
+                                            name!("biz"),
+                                            connect::fetch_dependency_graph::PathTailSelection::CustomScalarPathSelection {
+                                                path_selection: connect::fetch_dependency_graph::PathSelection::Selection(connect::SubSelection {
```

Comment:
should we make `SubSelection::parse` and the like crate-public so these tests are more concise?

Reply:
I'm not sure if this would help, since the `Path` input does not match 1:1 to any of the various `JSONSelection` types. Maybe a helper method to automatically construct a `Path` from a subselection would be useful, but let me know.

Reply:
Thanks. Fixed in https://github.com/apollographql/router/pull/5248